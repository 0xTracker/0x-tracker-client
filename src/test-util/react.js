import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
