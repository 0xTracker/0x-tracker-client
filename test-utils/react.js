import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRouter };
