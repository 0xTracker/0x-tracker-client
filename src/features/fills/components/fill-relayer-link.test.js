import React from 'react';

import { renderWithRouter } from '../../../../test-utils/react';
import FillRelayerLink from './fill-relayer-link';

it('returns relayer link when trade has a relayer', () => {
  const fill = {
    relayer: {
      name: 'Google',
      slug: 'google',
      url: 'https://google.com',
    },
  };
  const { container } = renderWithRouter(<FillRelayerLink fill={fill} />);

  expect(container.firstChild).toMatchSnapshot();
});

it('returns "None" when the trade does not have a fee recipient', () => {
  const fill = {
    feeRecipient: '0x0000000000000000000000000000000000000000',
  };
  const { container } = renderWithRouter(<FillRelayerLink fill={fill} />);

  expect(container.firstChild).toMatchSnapshot();
});

it('returns a fee recipient link when the trade does not have a relayer but does have a fee recipient', () => {
  const fill = {
    feeRecipient: '0x004e344251110fa1cb09aa31c95c6598ed07dce6',
  };
  const { container } = renderWithRouter(<FillRelayerLink fill={fill} />);

  expect(container.firstChild).toMatchSnapshot();
});
