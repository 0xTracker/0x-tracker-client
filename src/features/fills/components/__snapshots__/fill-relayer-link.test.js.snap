// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns "None" when the trade does not have a fee recipient 1`] = `None`;

exports[`returns a fee recipient link when the trade does not have a relayer but does have a fee recipient 1`] = `
<a
  class="link__StyledInternalLink-sc-1m1n17q-1 kIqcgN"
  href="/search?q=0x004e344251110fa1cb09aa31c95c6598ed07dce6"
>
  Unknown
</a>
`;

exports[`returns relayer link when trade has a relayer 1`] = `
<a
  class="link__StyledInternalLink-sc-1m1n17q-1 kIqcgN"
  href="/relayers/google"
>
  Google
</a>
`;
